sopel/irc/backends.py

```diff
@@ -127,7 +127,8 @@ def handle_connect(self):
     def handle_close(self):
         """Called when the socket is closed."""
         self.timeout_scheduler.stop()
-        self.timeout_scheduler.join(timeout=15)
+        if self.timeout_sheduler is not current_thread():
```

Comment:
```suggestion
        if current_thread() is not self.timeout_scheduler:
```

Note: Fixes a typo. The original line would not work (`self` has no attribute `timeout_sheduler`).

Reply:
agree

Reply:
agree